core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -951,6 +951,25 @@ wasi_path_remove_directory(wasm_exec_env_t exec_env, wasi_fd_t fd,
     return wasmtime_ssp_path_remove_directory(curfds, fd, path, path_len);
 }
 
+static __wasi_timestamp_t
+get_timeout_for_poll_oneoff(const wasi_subscription_t *in)
+{
+    __wasi_timestamp_t timeout = 0;
+    const __wasi_subscription_t *clock_subscription = NULL;
+    if (clock_subscription == NULL
```

Comment:
this condition is always true, since you set `clock_subscription` to `NULL` in the line above.